### PR TITLE
AbsSystem: Enable grabbing an absorption line

### PR DESCRIPTION
**Pull request by profxj**
Method to grab a named absorption line from an AbsSystem
tests
docs
notebook

### CODE REVIEW

File: linetools/isgm/abssystem.py [hunk 1]
```diff
@@ -183,6 +184,42 @@ def __init__(self, abs_type, radec, zabs, vlim, zem=0.,
         # Refs (list of references)
         self.Refs = []
 
+    def absline(self, inp):
```
Comment (ntejos) [inline linetools/isgm/abssystem.py]: Could you change the name to get_abslines() or something similar?

**Comment by profxj** [comment]
Renamed to get_absline()

File: docs/abssys.rst [hunk 2]
```diff
@@ -84,11 +84,30 @@ Plots
 Methods
 =======
 
+AbsLines
```
Comment (ntejos) [inline docs/abssys.rst]: Update name in the title too.

File: linetools/isgm/abscomponent.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ class AbsComponent(object):
         A comment, default is ``
     """
     @classmethod
-    def from_abslines(cls, abslines, **kwargs):
+    def from_abslines(cls, abslines, stars=None, **kwargs):
         """Instantiate from a list of AbsLine objects
 
         Parameters
```
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: add docstring for `stars` parameter

File: linetools/isgm/abssystem.py [hunk 4]
```diff
@@ -223,6 +224,42 @@ def chk_component(self, component):
...
+        -------
+        absline -- AbsLine object or list of Abslines
+          More than one will be returned if this line exists in
+          multiple components.  The returned quantity will then
+          be a list instead of a single object
+        """
+        # Generate the lines
+        abslines = self.list_of_abslines()
+        if isinstance(inp,basestring):
+            names = np.array([absline.trans for absline in abslines])
```
Comment (ntejos) [inline linetools/isgm/abssystem.py]: Should we change the attribute `trans` in AbsLines for `name` instead? (Not sure though)

**Comment by ntejos** [comment]
In general, I'm not so convinced about the `stars` parameter. I think it would be best to hard code the names of transitions so they include the asterisks when appropriate (the names should be unique of course).

File: docs/abssys.rst (hunk 2 above)
Comment (profxj) [inline docs/abssys.rst]: Not sure that makes sense.
Ignored.

File: linetools/isgm/abscomponent.py (hunk 3 above)
Comment (profxj) [inline linetools/isgm/abscomponent.py]: done

File: linetools/isgm/abssystem.py (hunk 4 above)
Comment (profxj) [inline linetools/isgm/abssystem.py]: Possibly, but not today.

**Comment by profxj** [comment]
The transitions are hard-coded.  But, it is possible to generate a component
without a transition and that is where this got difficult.

File: docs/abssys.rst (hunk 2 above)
Comment (ntejos) [inline docs/abssys.rst]: I thought you were naming sections as method names... I'm fine either way.

File: linetools/isgm/abscomponent.py [hunk 5]
```diff
@@ -152,6 +155,8 @@ def __init__(self, radec, Zion, z, vlim, Ej=0./u.cm, A=None, Ntup=None, comment=
             sig_logN : Error in log10 N
         Ej : Quantity, optional
             Energy of lower level (1/cm)
+        stars : str, optional
+            asterisks to add to name, e.g. '**' for CI**
```
Comment (ntejos) [inline linetools/isgm/abscomponent.py]: It seems that `stars` is only relevant when `name=None`. Please add this to the docstring.

**Comment by ntejos** [comment]
I see. Are you referring to creating AbsLines from `(Z,ion)`?

File: linetools/isgm/abssystem.py (hunk 4 above)
Comment (ntejos) [inline linetools/isgm/abssystem.py]: Will add a pull request shortly for addressing this.

File: linetools/isgm/abscomponent.py (hunk 5 above)
Comment (profxj) [inline linetools/isgm/abscomponent.py]: Added.

**Comment by profxj** [comment]
No, I mean when we generate AbsComponent with (Z,ion).

**Comment by profxj** [comment]
merging

**Pull request merged**